### ISSUE TITLE
Handle simpler tool input schemas for structured tools

### ISSUE THREAD

**Issue by ramnes** [langchain-ai/langchain#8066]
### Feature request

Recognize input schemas that do not use `Field` as structured schemas.

### Motivation

A simple schema without `Field` values such as this one will fail to be recognized as a structured schema:

```python
class RepeatTextSchema(BaseModel):
    text: str
    occurences: int
```

It will thus give such an error:

```
occurences
  field required (type=value_error.missing)
```

I guess it would provide a greater developer experience if any schema with multiple variables would be recognized as a structured schema.

### Your contribution

None so far but happy to help if I can.

**Comment by keenborder786** [comment]
@ramnes, let me see at this and see how I can make the feature a possibility.

**Comment by keenborder786** [comment]
@ramnes  I am not so sure what issue you are facing here, I tried passing structured schema to a tool without passing in Field Instance from pydantic

```python

from typing import Any, Dict

from langchain.agents import AgentType, initialize_agent
from langchain.llms import OpenAI
from langchain.tools import YouTubeSearchTool
from pydantic import BaseModel, Field, root_validator
import os


os.environ["OPENAI_API_KEY"] = ""
llm = OpenAI(model_name="gpt-3.5-turbo-16k",temperature=0)

class ToolInputSchema(BaseModel):
    query: str
    @root_validator
    def validate_query(cls, values: Dict[str, Any]) -> Dict:
        query = values["query"]
        if query[0] != 'L': raise ValueError(' Validation Check Failed')
        return values

tool = YouTubeSearchTool(args_schema=ToolInputSchema)

agent = initialize_agent(
    [tool], llm, agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION, verbose=False
)
answer = agent.run("Lex Friedman,5")
print(answer)

```

Please give more context so I can help you better.

**Comment by ramnes** [comment]
Thank you for taking a look into this @keenborder786. :)

As written in OP, the issue is about recognizing structured schemas, and thus structured tools. I just made the title more explicit. 

<details>
  <summary>This example works all the time.</summary>

```python
from typing import Type

from langchain.agents import AgentType, initialize_agent
from langchain.chat_models import ChatOpenAI
from langchain.tools import BaseTool
from pydantic import BaseModel, Field


class RepeatTextSchema(BaseModel):
    text: str = Field(default="", description="the text to repeat")
    occurences: int = Field(
        default=1, description="the number of times the text should be repeated"
    )


class RepeatTextTool(BaseTool):
    name = "RepeatText"
    description = "useful for when you need to repeat text a certain number of times"
    args_schema: Type[RepeatTextSchema] = RepeatTextSchema

    def _run(self, text, occurences):
        return " ".join([text] * occurences)

    async def _arun(self, text, occurences):
        return self._run(text, occurences)


llm = ChatOpenAI(temperature=0, streaming=True)
tools = [RepeatTextTool()]

agent = initialize_agent(
    tools=tools,
    agent=AgentType.STRUCTURED_CHAT_ZERO_SHOT_REACT_DESCRIPTION,
    llm=llm,
)
print(agent.run("Repeat this text 5 times: hello"))
```
</details>

<details>
  <summary>This one works sometimes, but regularly throws a validation error.</summary>

```python
from typing import Type

from langchain.agents import AgentType, initialize_agent
from langchain.chat_models import ChatOpenAI
from langchain.tools import BaseTool
from pydantic import BaseModel, Field


class RepeatTextSchema(BaseModel):
    text: str
    occurences: int


class RepeatTextTool(BaseTool):
    name = "RepeatText"
    description = "useful for when you need to repeat text a certain number of times"
    args_schema: Type[RepeatTextSchema] = RepeatTextSchema

    def _run(self, text, occurences):
        return " ".join([text] * occurences)

    async def _arun(self, text, occurences):
        return self._run(text, occurences)


llm = ChatOpenAI(temperature=0, streaming=True)
tools = [RepeatTextTool()]

agent = initialize_agent(
    tools=tools,
    agent=AgentType.STRUCTURED_CHAT_ZERO_SHOT_REACT_DESCRIPTION,
    llm=llm,
)
print(agent.run("Repeat this text 5 times: hello"))
```
</details>

<details>
  <summary>The complete error here.</summary>

```
Traceback (most recent call last):
  File "/app.py", line 34, in <module>
    print(agent.run("Repeat this text 5 times: hello"))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/langchain/chains/base.py", line 440, in run
    return self(args[0], callbacks=callbacks, tags=tags, metadata=metadata)[
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/langchain/chains/base.py", line 243, in __call__
    raise e
  File "/.venv/lib/python3.11/site-packages/langchain/chains/base.py", line 237, in __call__
    self._call(inputs, run_manager=run_manager)
  File "/.venv/lib/python3.11/site-packages/langchain/agents/agent.py", line 1010, in _call
    next_step_output = self._take_next_step(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/langchain/agents/agent.py", line 871, in _take_next_step
    observation = tool.run(
                  ^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/langchain/tools/base.py", line 267, in run
    parsed_input = self._parse_input(tool_input)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/langchain/tools/base.py", line 207, in _parse_input
    result = input_args.parse_obj(tool_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for RepeatTextSchema
occurences
  field required (type=value_error.missing)
```
</details>

**Comment by keenborder786** [comment]
Thank you for clarifying. I will get back to you, hopefully with a solution.

**Comment by keenborder786** [comment]
@ramnes, I've tested the code you provided several times and encountered the same error on some occasions. After investigating further, I discovered that the error occurs when the agent is deciding which tool to use based on the input. The output from the agent for that step looks like this:

```python
AgentAction(tool='RepeatText', tool_input={'text': 'hello', 'occurrences': 5}, log='Action: \n```\n{\n  "action": "RepeatText",\n  "action_input": {\n    "text": "hello",\n    "occurrences": 5\n  }\n}\n```')
```

The issue lies in the spelling of the parameter occurrences, which is different from what you specified as occurences. As a result, this inconsistency causes a parsing error. However, there are instances when the agent correctly predicts the output, which avoids the parsing error:

```python
AgentAction(tool='RepeatText', tool_input={'text': 'word', 'occurences': 2}, log='Action:\n```\n{\n  "action": "RepeatText",\n  "action_input": {\n    "text": "word",\n    "occurences": 2\n  }\n}\n```')
```

The reason the agent sometimes corrects the typo is due to the lack of explicit context provided through the Field description. As a result, the agent tries to infer the correct parameter name but occasionally gets it wrong, leading to the parsing error.

In conclusion, once I changed the misspelled occurences to occurrences, everything worked fine and the parsing error was resolved.

**Comment by ramnes** [comment]
Oh, that's very interesting, thanks! I really didn't see the typo until now. Do you think there's something to change in LangChain here or should I close the issue?

On one side this seems to be due to the typo mostly, which is an error on my side, but on the other side I feel like maybe there's something to enhance here, either the error message or the behavior when fields aren't described.

What do you think?

**Comment by hinthornw** [comment]
For the example provided, you should mark optional fields as Optional

**Comment by ramnes** [comment]
I don't get it, sorry. There is no optional argument in my tool here. What's your point?

**Comment by keenborder786** [comment]
@ramnes just mention in your prompt instructing the agent not to change the tool argument names and it should work.
